src/main/resources/templates/cypher/form.html

```diff
@@ -8,42 +8,99 @@
+            <div class="table-responsive">
+                <table class="table" style="font-size: 18px">
+                    <tr>
+                        <td>Id</td>
+                        <td><input th:field="*{cypherId}" readonly/></td>
+                    </tr>
+                    <tr>
+                        <td>Jméno</td>
+                        <td><input th:field="*{name}"/></td>
+                    </tr>
+                    <tr>
+                        <td>Pořadí</td>
+                        <td><input type="number" th:field="*{stage}"/></td>
+                    </tr>
+                    <tr>
+                        <td>Souřadnice</td>
+                        <td><input id="latLng" type="text" th:field="*{location}" readonly/></td>
+                    </tr>
+                    <tr>
+                        <td>Odpověď</td>
+                        <td><input type="text" th:field="*{codeword}"/></td>
+                    </tr>
+                    <tr>
+                        <td></td>
+                        <td><input type="submit" value="Uložit"/>
+                        </td>
+                    </tr>
+                </table>
+            </div>
+        </form>
+    </div>
+    <div id='map' style='width: 100%; height: 300px;'></div>
+    <script>
+        var map;
+        var activeMarker;
+
+        function initMap() {
+            var locationOfZlin = {lat: 49.22439, lng: 17.66485};
+
+            map = new google.maps.Map(document.getElementById('map'), {
+                zoom: 14,
+                center: locationOfZlin
+            });
+
+            map.addListener('click', function(event) {
+                deleteMarker();
+                addMarker(event.latLng);
+                fillLocationToForm(event);
+            });
+
+            addMarkerFromForm();
+        }
+
+        function addMarker(location) {
+            activeMarker = new google.maps.Marker({
+                position: location,
+                map: map
+            });
+        }
+
+        function fillLocationToForm(event) {
+            document.getElementById('latLng').value = event.latLng.lat() + ',' + event.latLng.lng();
+        }
+
+        function deleteMarker() {
+            if (activeMarker) {
+                activeMarker.setMap(null);
+            }
+        }
+
+        function addMarkerFromForm() {
+            var latlng = document.getElementById('latLng').value;
+            if (latlng) {
+                var toupleLatLng = latlng.split(",")
+                var lat = parseFloat(toupleLatLng[0]);
+                var lng = parseFloat(toupleLatLng[1]);
+                var latLngObject = {lat: lat, lng: lng};
+                addMarker(latLngObject)
+            }
+        }
+
+    </script>
+    <script async defer
+            src="https://maps.googleapis.com/maps/api/js?key=AIzaSyDseyIGXP_QJWmaCbjVJxxVUQ5vxX44Anw&callback=initMap">
+    </script>
```

Comment:
nemam slov, vypada to uzasne, vic tomu nerozumim. Ciste html-ko jsem si dala do notepad - a funguje. Tomu ostatnimu verim :-). Za me - palec nahoru